doxygen/dox/Unicode.dox

```diff
@@ -0,0 +1,144 @@
+
+\section sec_unicode_warn Caveats, Pitfalls, and Things to Watch For
+Programmers who are accustomed to using ASCII text without accommodating other text
+encodings will have to be aware of certain common issues as they begin using UTF-8 encodings.
+
+\subsection subsec_unicode_warn_port Cross-platform Portability
+Since the HDF5 Library handles datatypes directly, UTF-8 encoded text in dataset and
+attribute datatypes in a well-designed HDF5 application and file should work transparently
+across platforms. The same should be true of handling names of groups, datasets, committed
+datatypes, and attributes within a file.
+
+Be aware, however, of system or application limitations once data or other information
+has been extracted from an HDF5 file. The application or system must be designed to
+accommodate UTF-8 encodings if the information is then used elsewhere in the application or system environment.
+
+Data from a UTF-8 encoded HDF5 datatype, in either a dataset or an attribute,
+that has been established within an HDF5 application should "just work" within the HDF5 portions of the application.
+
+\subsection subsec_unicode_warn_names Filenames
+Since file access is a system issue, filenames do not fall within the scope
+of HDF5's UTF-8 capabilities; filenames are encoded at the system level.
+
+Linux and Mac OS systems normally handle UTF-8 encoded filenames correctly
+while Windows systems generally do not.
+
+\section sec_unicode_text The *Plain Text* Illusion
+Beware the use of the term *plain text*. *Plain text* is at best ambiguous, but often
+misleading. Many will assume that *plain text* means ASCII, but plain text German or
+French, for example, cannot be represented in ASCII. Plain text is only unambiguous
+in the context of English (and even then can be problematic!).
+
+\subsection subsec_unicode_warn_store Storage Size
+Programmers and data users accustomed to working strictly with ASCII data generally make
+the reasonable assumption that 1 character, be it in an object name or in data, requires
+1 byte of storage. This equation does not work when using UTF-8 or any other Unicode encoding.
+With Unicode encoding, number of characters is not synonymous with number of bytes. One must
+get used to thinking in terms of number of characters when talking about content, reserving
+number of bytes for discussions of storage size.
+
+When working with Unicode text, one can no longer assume a 1:1 correspondence between the
+number of characters and the data storage requirement.
+
+\subsection subsec_unicode_warn_sys System Dependencies
+<b>Linux</b>, <b>Unix</b>, and similar systems generally handle UTF-8 encodings in correct and
+predictable ways. There is an apparent consensus in the Linux community that "UTF-8 is just the right way to go."
+
+<b>Mac OS</b> systems generally handle UTF-8 encodings correctly.
+
+<b>Windows</b> systems use a different Unicode encoding, UCS-2 (discussed in this UTF-16 article) at
```

Comment:
No article linked here. Probably supposed to link to https://en.wikipedia.org/wiki/UTF-16